trie/statistics/trieStatistics.go

```diff
@@ -101,12 +106,13 @@ type TrieStatsDTO struct {
 	TotalNumNodes  uint64
 	MaxTrieDepth   uint32
 
-	BranchNodesSize    uint64
-	NumBranchNodes     uint64
-	ExtensionNodesSize uint64
-	NumExtensionNodes  uint64
-	LeafNodesSize      uint64
-	NumLeafNodes       uint64
+	BranchNodesSize      uint64
+	NumBranchNodes       uint64
+	ExtensionNodesSize   uint64
+	NumExtensionNodes    uint64
+	LeafNodesSize        uint64
+	NumLeafNodes         uint64
+	LeavesMigrationStats map[core.TrieNodeVersion]uint64
```

Comment:
a little bit dangerous to use a map in a plain DTO.
Reason: this DTO can be passed among multiple structs. If we happen to do a write in the map while executing a read, a panic can occur. For the other fields, the code won't panic event if the concurrency issue happens. Suggestion to refactor the solution.

Reply:
Refactored.